tests/array_test.py

```diff
@@ -730,7 +730,7 @@ def test_devices_sharding_op_sharding_lowering(
     mesh = jtu.create_global_mesh((4, 2), ('x', 'y'))
     mps = sharding.MeshPspecSharding(mesh, pspec)
 
-    devices_sharding = sharding.DevicesSharding(jax.devices())
+    devices_sharding = sharding.ReshapeableDevicesSharding(jax.devices())
     devices_sharding = devices_sharding.reshape(shape).replicate(axes)
```

Comment:
Maybe add a test where you don't do this reshape? To test that you are actually respecting the user shape?

In this test, you can get it by doing `mesh.devices` to get a shape that's not 1D

Reply:
Done!